include/matx/transforms/reduce.h

```diff
@@ -1269,6 +1268,53 @@ void __MATX_INLINE__ sum(OutType dest, const InType &in, cudaStream_t stream = 0
 #endif  
 }
 
+template <int RANK, class T>
+auto __MATX_INLINE__ getPermuteDims( T &dims ) {
+  std::array<int, RANK> perm;
+  std::array<bool, RANK> visited;
+
+  visited.fill(false);
+  
+  // construct permutation array by moving fastest changing dims to end
+  int j = RANK-1;
+  for(auto it = rbegin(dims); it != rend(dims); it++) {
+    int a = *it;
+    MATX_ASSERT_STR(a >= 0 && a < RANK, matxInvalidDim, "Reduction dim out of range\n");
+    MATX_ASSERT_STR(visited[a] == false, matxInvalidDim, "Reduction Dim repeated");
+
+    visited[a] = true;
+
+    perm[j--] = a;
+  }
+
+  // now add remaning dims to front
+  j = 0;
+  for(int i = 0; i < RANK;  i++) {
+    if(!visited[i]) {
+      perm[j++] = i;
+    }
+  }
+
+  return perm;
+}
+
+template <typename OutType, typename InType>
+void __MATX_INLINE__ sum(OutType dest, const InType &in, std::initializer_list<int> dims, cudaStream_t stream = 0)
```

Comment:
should use `const index_t (&arr)[R]` instead I think with R as a template value type?

Reply:
I'm not sure I follow.  Could you explain more?

Reply:
Instead of `initializer_list`, which has other problems, we should use a C-array so it can use the `{}` syntax but also allow arrays to be passed in like the other functions.

Reply:
we have the array APIs also.  Maybe initializer_list isn't correct.  Perhaps a vector would be better?

Reply:
now using vector.  if acceptable please resolve this conversation.

Reply:
I think I might be missing context here, but all of these APIs should be using `const index_t (&arr)[R]` and not vector or initializer_list. vectors have extra overhead that isn't needed (dynamic allocations), and initializer_lists have their own set of problems. I think you can just replace all the instances of vector with C arrays, ie:

```
template <typename OutType, typename InType>
void __MATX_INLINE__ stdd(OutType dest, const InType &in, std::vector<int> dims, cudaStream_t stream = 0)
```
becomes
```
template <typename OutType, typename InType, int R>
void __MATX_INLINE__ stdd(OutType dest, const InType &in, const int (&dims)[R], cudaStream_t stream = 0)
```

Reply:
not for this parameter as the size of R is not necessarily known at compile time.  We know it is <= dims.  We could add another template parameter as you do above and force it throughout the APIs if we deem that better but I think it is honestly unnecessary.

Reply:
fixed